roles/kubernetes/control-plane/tasks/kubeadm-setup.yml

```diff
@@ -183,13 +183,32 @@
   environment:
     PATH: "{{ bin_dir }}:{{ ansible_env.PATH }}"
   notify: Master | restart kubelet
+  
+  # The following variable contains another resolve pattern
+  # kubeadm_certificate_key:  {{ lookup('password', credentials_dir + '/kubeadm_certificate_key.creds length=64 chars=hexdigits') | lower }}
+  # if you have MORE than one master/control-plane, then this creates a race condition
+  # where the lookup tries to lock the "kubeadm_certificate_key.creds", which leads to
+  # an exception.
+  #
+  # Avoid the race condition by running the resolve only once before entering the step "set kubeadm"
+  # because this is a lookup to a static filename, which does not change
+  
+- name: set kubeadm_certificate_key_resolved to avoid race condition
+  set_fact:
+    kubeadm_certificate_key_resolved: "{{ kubeadm_certificate_key }}"
+  run_once: true
+
+- name: set kubeadm_certificate_key_resolved to avoid race condition
```

Comment:
has the job been duplicated?

Reply:
oh the name might be misleading. the first step is "preparing the key" in an ansible variable (198)  - no changes to the system. The Second is setting the key aka making a change to the system (201).
Not sure if there a better ways to explain it in short